### PR TITLE
Fix input autofocus position

### PR DESCRIPTION
**Pull request by mmarkelov**
https://github.com/facebook/react/issues/14125

I am not sure that it is right approach. So it will be fine, if someone could check it. Maybe I miss something.

### CODE REVIEW

File: packages/react-dom/src/__tests__/ReactDOM-test.js [hunk 1]
```diff
@@ -308,14 +308,15 @@ describe('ReactDOM', () => {
...
-          <input autoFocus={true} />
+          <input defaultValue="foo" autoFocus={true} />
           <p>The above input should be focused after mount.</p>
         </div>,
         container,
       );
 
       expect(inputFocusedAfterMount).toBe(true);
       expect(focusedElement.tagName).toBe('INPUT');
+      expect(focusedElement.selectionStart).toBe(0);
```
Comment (Matthew-Goldberg) [inline packages/react-dom/src/__tests__/ReactDOM-test.js]: Are you sure this test is useful?  It seems this assertion passes even without a fix to the actual problem.

**Comment by xavierrutayisire** [comment]
Any news about this PR?
It would be nice to have this fix.

**Comment by xavierrutayisire** [comment]
Hey, do you know if we could have this fix in an upcoming release?
I'm working on a project where the wrong autofocus position is a blocking problem.

**Comment by xavierrutayisire** [comment]
Hi @mmarkelov @vikaskyadav @Matthew-Goldberg,

Do you know if this pull request can still be done soon?

**Comment by Matthew-Goldberg** [comment]
The challenge was figuring out how to write a test; it seems jsdom is showing that the selectionStart is 0 even though in the browser the same code shows selectionStart is the end/length of the text value.  Maybe an issue with jsdom.

**Comment by Matthew-Goldberg** [comment]
I have opened an issue with `jsdom` to see if we can get that library updated so we could write a valid test.
https://github.com/jsdom/jsdom/issues/2787

**Comment by Matthew-Goldberg** [comment]
OK the `jsdom` issue is now fixed: https://github.com/jsdom/jsdom/pull/2799
When this is released we can upgrade the `jsdom` version to pick up the fix, and re-implement the test which should now be useful.

**Comment by Matthew-Goldberg** [comment]
`jsdom` fix is released https://github.com/jsdom/jsdom/releases/tag/16.1.0

**Comment by gaearon** [comment]
We're going to remove attribute syncing in one of the next majors. Then this will be solved. I don't think we should be adding more complexity here because it is already a lot.

**Comment by Matthew-Goldberg** [comment]
@gaearon it may still be useful to implement the test @mapkeji added in 9d78fbe54a1c465bb2a36951ead5d8d91a73d8e3
At the time, this test would pass either way so it was not useful.
With the `jsdom` upgrade it should be useful.
In any case, can't add the working test unless the actual implementation matches spec.
But just a note in case the expectation is to restore this behavior with the attribute syncing change.

**Pull request closed without merging**